### PR TITLE
Reproductive Rights

### PR DESCRIPTION
**Pull request by CodaFi**
A partial solution to #116.  Next step is to expose the `Result` type and give it a better name.

### CODE REVIEW

File: SwiftCheck/Arbitrary.swift [hunk 1]
```diff
@@ -146,7 +144,7 @@ extension Int64 : Arbitrary {
 
 extension UInt : Arbitrary {
 	public static var arbitrary : Gen<UInt> {
-		return Gen.sized { n in Gen<UInt>.pure(UInt(arc4random_uniform(UInt32(abs(n))))) }
+		return Gen.sized { n in Gen<UInt64>.choose((0, UInt(n))) }
```
Comment (mpurland) [inline SwiftCheck/Arbitrary.swift]: Should this be Gen<UInt>?

Comment (CodaFi) [inline SwiftCheck/Arbitrary.swift]: Yep! Good catch.

File: SwiftCheck/Arbitrary.swift [hunk 2]
```diff
@@ -198,12 +196,20 @@ extension UInt64 : Arbitrary {
 
 extension Float : Arbitrary {
 	public static var arbitrary : Gen<Float> {
-		return Gen.sized({ n in
+		#if !(arch(x86_64) || arch(arm64))
+			let precision : Int = 999999999
```
Comment (mpurland) [inline SwiftCheck/Arbitrary.swift]: Would be nice if this was defined as a constant somewhere.

File: SwiftCheck/Arbitrary.swift (hunk 1 above)
Comment (mpurland) [inline SwiftCheck/Arbitrary.swift]: :+1:

**Comment by CodaFi** [comment]
:boat:

**Comment by mpurland** [comment]
:+1:

**Pull request merged**